### DISCUSSION TITLE
Disabling allow-list for administrator dashboard users

### DISCUSSION THREAD

**Discussion by gradisarjoze** [hasura/graphql-engine#8214] [General] [answered]
At the moment we are using Hasura in production. We have set up automatic allowlist generation and enabled it for our application.
Now we are trying to create an admin dashboard so a team can manage a few specific parts of our app.

To implement the admin dashboard we are looking at React Admin and `ra-data-hasura` package.

We would create a new role - `administrator` which would have specific permissions so the administrator users can modify some of the data.

How can we have allow-list enabled for the application and perhaps modified/disabled for the administrator dashboard?

An option would be creating a separate allowlist for `administrator` role but it would be a pretty manual process and it would need to be modified manually every time we update the dashboard or the database structure.

Is there a way to e.g. enable `administrator` role to bypass the allowlists like the `admin` role does?

**Comment by BenoitRanque** [comment, marked as answer]
Selective application of the allow list is not currently supported. Feel free to open a feature request for this feature.

If you want to generate the allow list dynamically based on the queries in your dashboard, you can can take a look at the [hasura allow list graphql-codegen plugin](https://www.graphql-code-generator.com/plugins/hasura-allow-list). But this won't solve the issue of only allowing some roles to use those queries.
